File: BE/M2/BW2/EnergyService/src/main/java/com/BEBW2/ES/EnergyService/Entities/Address.java
package com.BEBW2.ES.EnergyService.Entities;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
@Table(name = "addresses")
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String via;

    private int civico;

    private int cap;

    @ManyToOne
    private Comune comune;

    public Address(String via, int civico, int cap, Comune comune){

        this.via = via;
        this.civico = civico;
        //TODO chiedere localita
        this.cap=cap;
        this.comune= comune;
    }

}

File: BE/M2/BW2/EnergyService FE/src/app/models/address.ts
import { Comune } from "./comune";

export class Address {
  id!:number;
  via:string;
  civico:number;
  cap:number;
  comune:Comune;

  constructor(via:string, civico:number, cap:number, comune:Comune) {
    this.via = via;
    this.civico = civico;
    this.cap = cap;
    this.comune = comune;
  }
}
